inc/zoo/swar/SWAR.h

```diff
@@ -62,13 +74,53 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+            if constexpr (NBits < sizeof(T) * 8) {
+                return (T(1) << NBits) - 1;
+            } else {
+                return ~T(0);
+            }
+        }(),
         // Use LowerBits in favor of ~MostSignificantBit to not pollute
         // "don't care" bits when non-power-of-two bit lane sizes are supported
-        LowerBits = MostSignificantBit - LeastSignificantBit;
+        LowerBits = MostSignificantBit - LeastSignificantBit,
+        MaxUnsignedLaneValue = LeastSignificantLaneMask;
+
+    template <typename InputIt>
+    constexpr static auto from_range(InputIt first, InputIt last) noexcept {
```

Comment:
is `from_range` a developing consensus concerning constructors from ranges?
I think `range` is not necessary.
Also, I am suspicious of these pseudo-constructors (you need to call them from the type as in `SWAR<yada yada>::from_range)` these are like something we can call "named constructors".  Do you have inspiration from some other code about this idiom?

Reply:
I feel like I've seen a few patterns like this, especially in unit conversion. 
I think it reads nicely. 

It could be valuable for you to create a supplementary PR for this to: 
- make this more real constructor centric
- do the compile-time index_sequence impl

what do you think ?